contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,338 @@
+        uint8 move
+    );
+
+    event DrawRefund(
+        address indexed playerOne,
+        address indexed playerTwo,
+        bytes32 indexed gameKey,
+        uint256 playerOneRefund,
+        uint256 playerTwoRefund
+    );
+
+    event GamePayment(
+        address indexed player,
+        bytes32 indexed gameKey,
+        uint256 amount
+    );
+
+    event ForfeitPaid(
+        address indexed player,
+        bytes32 indexed gameKey,
+        uint256 amount
+    );
+
+    event WithDraw(
+        address indexed player,
+        uint amount
+    );
+
+    event PlayerOneEndsGame(
+        address indexed player,
+        bytes32 indexed gameKey,
+        uint256 amount
+    );
+
+    // The creator's move is embedded in the game key
+    function movePlayerOne(bytes32 moveHash, uint256 stake) external payable whenNotPaused {
+
+        require(moveHash != bytes32(0), "Move hash is empty");
+        Game storage game = games[moveHash];
+        require(game.step == uint8(Steps.INIT), INVALID_STEP_MSG);
+        require(game.playerOne == NULL_ADDRESS, INVALID_MOVE_MSG);
+
+        game.stake = stake;
+        game.playerOne = msg.sender;
+        game.step = uint8(Steps.PLAYER_ONE_MOVE);
+        depositStake(stake);
+
+        emit PlayerOneMoves(msg.sender, moveHash, stake, msg.value);
+    }
+
+    function movePlayerTwo(bytes32 gameKey, bytes32 moveHash) external payable whenNotPaused {
+
+        require(moveHash != bytes32(0), "Move hash is empty");
+        Game storage game = games[gameKey];
+        require(game.step == uint8(Steps.PLAYER_ONE_MOVE), INVALID_STEP_MSG);
+
+        uint256 expiryDate = block.timestamp.add(FORFEIT_WINDOW);
+        game.playerTwo = msg.sender;
+        game.playerTwoMoveHash = moveHash;
+        game.step = uint8(Steps.PLAYER_TWO_MOVE);
+        game.expiryDate = expiryDate;
+        depositStake(game.stake);
+
+        emit PlayerTwoMoves(msg.sender, msg.value, gameKey, moveHash, expiryDate);
+    }
+
+    function depositStake(uint256 stake) internal {
+
+        if (msg.value == stake) {
+            return;
+        }
+
+        uint256 senderBalance = balances[msg.sender];
+        if (msg.value < stake) {
+            require(senderBalance >= stake.sub(msg.value), "Insufficient balance");
+        }
+
+        balances[msg.sender] = senderBalance.add(msg.value).sub(stake);
+    }
+
+    function revealPlayerOne(bytes32 secret, uint256 playerOneMove) external whenNotPaused {
+
+        require(playerOneMove > 0 && playerOneMove < 4, INVALID_MOVE_MSG);
+        bytes32 gameKey = createPlayerOneMoveHash(msg.sender, secret, playerOneMove);
+        Game storage game = games[gameKey];
+        require(game.step == uint8(Steps.PLAYER_TWO_MOVE), INVALID_STEP_MSG);
+
+        uint256 expiryDate = block.timestamp.add(FORFEIT_WINDOW);
+        game.playerOneMove = uint8(playerOneMove);
+        game.step = uint8(Steps.PLAYER_ONE_REVEAL);
+        game.expiryDate = expiryDate;
```

Comment:
You are referring to unnecessary casting to `uint8()`?

`game.step = uint8(Steps.PLAYER_ONE_REVEAL);`

Reply:
I am talking about

- doing something about the step.
- doing something with expiry.
- doing something about the step.
- doing something with expiry.

Instead of:

- doing something about the step.
- doing something about the step.
- doing something with expiry.
- doing something with expiry.

Reply:
I understand. I realised when I saw your other comment. I will look out for that issue.